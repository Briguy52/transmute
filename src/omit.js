import curry from "./curry";
import { Seq } from "immutable";
import filterNot from "./filterNot";

const filterNotOperation = filterNot.operation;

function omit(keys, subject) {
  const keySet = Seq.Set(keys);
  return filterNotOperation((value, key) => keySet.contains(key), subject);
}

/**
 * Drop specified keys from a KeyedIterable (e.g. a `Map` or `OrderedMap`).
 *
 * @example
 * // returns Map { "two" => 2 }
 * omit(
 *   ['one', 'three'],
 *   Map({one: 1, two: 2, three: 3})
 * );
 *
 * @param  {Array<any>} keys to remove.
 * @param  {KeyedIterable} subject from which to remove `keys`.
 * @return {KeyedIterable} without `keys`.
 */
export default curry(omit);
