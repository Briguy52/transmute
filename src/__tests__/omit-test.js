import { Map, Record } from "immutable";
import omit from "../omit";

describe("transmute/omit", () => {
  it("drops keys from Maps", () => {
    expect(
      omit(["one", "three"], Map({ one: 1, two: 2, three: 3 }))
    ).toMatchSnapshot();
  });

  it("drops keys from Records", () => {
    const TestRecord = Record(
      {
        one: 1,
        two: 2,
        three: 3
      },
      "TestRecord"
    );
    expect(omit(["one", "three"], TestRecord())).toMatchSnapshot();
  });
});
