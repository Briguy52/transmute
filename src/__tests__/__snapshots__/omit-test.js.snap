// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/omit drops keys from Maps 1`] = `
Immutable.Map {
  two: 2,
}
`;

exports[`transmute/omit drops keys from Objects 1`] = `
Object {
  "two": 2,
}
`;

exports[`transmute/omit drops keys from Records 1`] = `
Immutable.Map {
  two: 2,
}
`;
